agent.go

```diff
@@ -229,18 +234,22 @@ func (agent *agentS) head(url string) (string, error) {
 }
 
 func (agent *agentS) request(url string, method string, data interface{}) (string, error) {
-	return agent.fullRequestResponse(url, method, data, nil, "")
+	ctx, cancel := context.WithTimeout(context.Background(), agent.clientTimeout)
```

Comment:
Since it's not clear from the method name that these methods are used for the data transmission cycle, shall we put a comment that explains why there is a different timeout here?